src/authTokenInterceptor.ts

```diff
@@ -182,33 +148,30 @@ export const authTokenInterceptor = ({
   return async (requestConfig: any): Promise<any> => {
     // Waiting for a fix in axios types
     // We need refresh token to do any authenticated requests
-    if (!getRefreshToken()) return requestConfig
-
-    // Queue the request if another refresh request is currently happening
-    if (isRefreshing) {
-      return new Promise((resolve, reject) => {
-        queue.push({ resolve, reject })
-      })
-        .then((token) => {
-          if (requestConfig.headers) {
-            requestConfig.headers[header] = `${headerPrefix}${token}`
-          }
-          return requestConfig
-        })
-        .catch(Promise.reject)
-    }
-
-    // Do refresh if needed
-    let accessToken
-    try {
-      accessToken = await refreshTokenIfNeeded(requestRefresh)
-      resolveQueue(accessToken)
-    } catch (error: unknown) {
-      if (error instanceof Error) {
-        declineQueue(error)
-        throw new Error(
-          `Unable to refresh access token for request due to token refresh error: ${error.message}`
-        )
+    if (!(await getRefreshToken())) return requestConfig
+
+    let accessToken = undefined
+
+    // Try to await a current request
+    if (currentlyRequestingPromise) accessToken = await currentlyRequestingPromise
+
+    if (!accessToken) {
+      try {
+        // Sets the promise so everyone else will wait - then get the value
+        currentlyRequestingPromise = refreshTokenIfNeeded(requestRefresh)
+        accessToken = await currentlyRequestingPromise
+
+        // Reset the promise
+        currentlyRequestingPromise = undefined
+      } catch (error: unknown) {
+        // Reset the promise
+        currentlyRequestingPromise = undefined
+
```

Comment:
Can put this in a `finally` block, it will [always get called](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)

Reply:
I don't understand?  No need for `finally`... we're using `await`.

Reply:
Wait - I see what you are saying now.  I thought you were referencing waiting on the Promise - but you were actually talking about setting it to `undefined`.  Sure - that would be fine.

I don't actually know the semantics though - does a `finally` get run even if we rethrow like we are on 171?  At any rate - I would say to hold off on that change until we get this in.  What's here now matches up with the original code.